confignode/src/main/java/org/apache/iotdb/confignode/manager/TriggerManager.java

```diff
@@ -0,0 +1,96 @@
+ * under the License.
+ */
+
+package org.apache.iotdb.confignode.manager;
+
+import org.apache.iotdb.common.rpc.thrift.TDataNodeLocation;
+import org.apache.iotdb.common.rpc.thrift.TSStatus;
+import org.apache.iotdb.commons.path.PartialPath;
+import org.apache.iotdb.commons.path.PathDeserializeUtil;
+import org.apache.iotdb.commons.trigger.TriggerInformation;
+import org.apache.iotdb.confignode.consensus.request.read.GetTriggerTablePlan;
+import org.apache.iotdb.confignode.consensus.response.TriggerTableResp;
+import org.apache.iotdb.confignode.persistence.TriggerInfo;
+import org.apache.iotdb.confignode.rpc.thrift.TCreateTriggerReq;
+import org.apache.iotdb.confignode.rpc.thrift.TDropTriggerReq;
+import org.apache.iotdb.confignode.rpc.thrift.TGetTriggerTableResp;
+import org.apache.iotdb.confignode.rpc.thrift.TTriggerState;
+import org.apache.iotdb.rpc.TSStatusCode;
+import org.apache.iotdb.trigger.api.enums.TriggerEvent;
+import org.apache.iotdb.trigger.api.enums.TriggerType;
+import org.apache.iotdb.tsfile.utils.Binary;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+public class TriggerManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TriggerManager.class);
+
+  private final ConfigManager configManager;
+  private final TriggerInfo triggerInfo;
+
+  public TriggerManager(ConfigManager configManager, TriggerInfo triggerInfo) {
+    this.configManager = configManager;
+    this.triggerInfo = triggerInfo;
+  }
+
+  public TriggerInfo getTriggerInfo() {
+    return triggerInfo;
+  }
+
+  public TSStatus createTrigger(TCreateTriggerReq req) {
+    boolean isStateful = TriggerType.construct(req.getTriggerType()) == TriggerType.STATEFUL;
+    TDataNodeLocation dataNodeLocation =
+        isStateful ? configManager.getNodeManager().getLowestLoadDataNode() : null;
+    TriggerInformation triggerInformation =
+        new TriggerInformation(
+            (PartialPath) PathDeserializeUtil.deserialize(req.pathPattern),
+            req.getTriggerName(),
+            req.getClassName(),
+            req.getJarPath(),
+            req.getAttributes(),
+            TriggerEvent.construct(req.triggerEvent),
+            TTriggerState.INACTIVE,
+            isStateful,
+            dataNodeLocation,
+            req.getJarMD5());
+    return configManager
+        .getProcedureManager()
+        .createTrigger(triggerInformation, new Binary(req.getJarFile()));
+  }
+
+  public TSStatus dropTrigger(TDropTriggerReq req) {
+    // TODO
+    return null;
+  }
+
+  public TGetTriggerTableResp getTriggerTable() {
+    try {
+      return ((TriggerTableResp)
+              configManager.getConsensusManager().read(new GetTriggerTablePlan()).getDataset())
+          .convertToThriftResponse();
+    } catch (IOException e) {
+      e.printStackTrace();
+      return new TGetTriggerTableResp(
+          new TSStatus(TSStatusCode.EXECUTE_STATEMENT_ERROR.getStatusCode()), null);
```

Comment:
It's better to use the e.message() as the response TSStatus's message. Therefore the client that connects to DataNode will receive the reason why cause Exception.